grpc_opa/authorizer.go

```diff
@@ -343,3 +345,62 @@ func addObligations(ctx context.Context, opaResp OPAResponse) (context.Context,
 	}
 	return ctx, err
 }
+
+// NewAuthorizationConfigUnary returns new unary config for authorization requests
+func NewAuthorizationConfigUnary(application string, opts ...Option) *Config {
+	cfg := &Config{
+		address: opa_client.DefaultAddress,
+	}
+
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	if cfg.authorizer == nil {
+		log.Info("authorizers empty, using default authorizer")
+		cfg.authorizer = []Authorizer{NewDefaultAuthorizer(application, opts...)}
+	}
+
+	return cfg
+}
+
+// AffirmAuthorizationUnary makes an authz request to sidecar-OPA.
+// If authorization is permitted, error returned is nil,
+// and a new context is returned containing any obligations
+// which caller must further evaluate if required.
+func (cfg *Config) AffirmAuthorizationUnary(ctx context.Context, fullMethod string, grpcReq interface{}) (context.Context, error) {
+	logger := ctxlogrus.Extract(ctx)
+	var (
+		ok     bool
+		newCtx context.Context
+		err    error
+	)
+
+	if cfg.authorizer == nil {
+		err = ErrNoAuthorizer
+		logger.WithError(err).Error(err.Error())
+		return nil, err
+	}
```

Comment:
Inject the default authorizer if none was specific, like it was before.

Reply:
No we can't inject default authorizer here because the opts parm is not available.
Anyways I think it's better to return error because it indicates to caller that Config was not init'd properly.

Reply:
opts are in the config, this should not be a method off of config